accord-core/src/main/java/accord/impl/AbstractConfigurationService.java

```diff
@@ -45,15 +48,16 @@ public abstract class AbstractConfigurationService<EpochState extends AbstractCo
 
     protected final EpochHistory epochs = createEpochHistory();
 
-    protected final List<Listener> listeners = new ArrayList<>();
+    protected final List<Listener> listeners = new CopyOnWriteArrayList<>();
```

Comment:
there are now 2 different types of locking in this class: `listeners` that use the copy-on-write, and `epochs` which does sync on the class.

One of the reasons to make this `CopyOnWrite` is that its accessed by different threads (register, caller) and this is a simple way to make sure its safe without needing this class to manage the locks for it (before we did class level sync)